crates/ruff_python_ast/src/node.rs

```diff
@@ -3801,12 +3797,11 @@ impl AstNode for ast::PatternMatchMapping {
         V: PreorderVisitor<'a> + ?Sized,
     {
         let ast::PatternMatchMapping {
-            keys,
-            patterns,
+            items,
             range: _,
             rest: _,
         } = self;
-        for (key, pattern) in keys.iter().zip(patterns) {
+        for ast::MatchMappingItem { key, pattern } in items {
             visitor.visit_expr(key);
             visitor.visit_pattern(pattern);
         }
```

Comment:
Hmm, here I didn't change the iteration order, but according to the results of `dis.dis()` in https://github.com/astral-sh/ruff/pull/11267?notification_referrer_id=NT_kwDOA_A9dbQxMDUxODQ2OTEwNTo2NjA3NjAyMQ#issuecomment-2096090737, the existing order might be incorrect? Possibly we should be visiting all the keys in a `PatternMatchMapping` before we visit any of the values, since it seems (unlike with dictionary literals) the Python runtime loads all the keys before loading any of the values when it comes to mapping patterns